kolibri/content/api.py

```diff
@@ -4,221 +4,173 @@
 """
 from functools import wraps
 
-from django.core.files import File as DjFile
 from django.db.models import Q
 from kolibri.content import models as KolibriContent
 from kolibri.content.utils import validate
+from .constants import content_kinds
 
 """ContentDB API methods"""
 
 def can_get_content_with_id(func):
     """
-    decorator function for returning ContentMetadata object when giving content id.
+    decorator function for returning ContentNode object when giving content id.
     it can take keyword argument/s "content" or "content1" and "content2".
     """
     @wraps(func)
     def wrapper(channel_id=None, **kwargs):
         content = kwargs.get('content')
         content1 = kwargs.get('content1')
         content2 = kwargs.get('content2')
+        target_node = kwargs.get('target_node')
+        prerequisite = kwargs.get('prerequisite')
 
-        if isinstance(content, KolibriContent.ContentMetadata) or \
-                (isinstance(content1, KolibriContent.ContentMetadata) and isinstance(content2, KolibriContent.ContentMetadata)):
+        if isinstance(content, KolibriContent.ContentNode) or \
+                (isinstance(content1, KolibriContent.ContentNode) and isinstance(content2, KolibriContent.ContentNode)) or \
+                (isinstance(target_node, KolibriContent.ContentNode) and isinstance(prerequisite, KolibriContent.ContentNode)):
             pass
         elif validate.is_valid_uuid(content):
-            kwargs['content'] = KolibriContent.ContentMetadata.objects.using(channel_id).get(content_id=content)
+            kwargs['content'] = KolibriContent.ContentNode.objects.using(channel_id).get(content_id=content)
         elif validate.is_valid_uuid(content1) and validate.is_valid_uuid(content2):
-            kwargs['content1'] = KolibriContent.ContentMetadata.objects.using(channel_id).get(content_id=content1)
-            kwargs['content2'] = KolibriContent.ContentMetadata.objects.using(channel_id).get(content_id=content2)
+            kwargs['content1'] = KolibriContent.ContentNode.objects.using(channel_id).get(content_id=content1)
+            kwargs['content2'] = KolibriContent.ContentNode.objects.using(channel_id).get(content_id=content2)
+        elif validate.is_valid_uuid(target_node) and validate.is_valid_uuid(prerequisite):
+            kwargs['target_node'] = KolibriContent.ContentNode.objects.using(channel_id).get(content_id=target_node)
+            kwargs['prerequisite'] = KolibriContent.ContentNode.objects.using(channel_id).get(content_id=prerequisite)
         else:
-            raise TypeError("must provide a ContentMetadata object or a UUID content_id")
+            raise TypeError("must provide a ContentNode object or a UUID content_id")
         return func(channel_id=channel_id, **kwargs)
     return wrapper
 
 def get_content_with_id(channel_id=None, content=None):
```

Comment:
Looks like both of these parameters are required. They shouldn't be keyword arguments then :)